example.R

```diff
@@ -16,44 +16,44 @@ shared <- function(auth)
 
 one_drive_2_R <- function(auth, path, use.readr=FALSE, FUN=NULL, ...)
 {
-  items        <- strsplit(path, "/")[[1]]
-  subdirs      <- items[-c(1, length(items))]
+  filename     <- basename(path)
+  items        <- strsplit(dirname(path), "/")[[1]]
+  path_len     <- length(items)
+  # case: "path" is top-level filename
+  if(path_len == 1 && items == '.')
+  {
+    items <- fp <- filename
+  } else
+  {
+    fp <- do.call(file.path, as.list(c(items[-1], filename)))
```

Comment:
I don't understand why this is needed. Wouldn't `fp <- as.list(c(items[-1], filename))` work just the same? There is no "relative" expansions allowed traversing a MS directory.